### PR TITLE
Extract GHA From #37

### PR DESCRIPTION
**Pull request by hexylena**
We clearly needed this sooner.

I am morally opposed to `isort` linting, if there is something that will automatically apply the `isort`, then awesome, let's add it to the `reformat` make target, but the linting without an automatic fix I object to.

### CODE REVIEW

**Comment by hexylena** [comment]
@bernt-matthias look ok to you?

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    - name: install packages for testing
+      run: pip install nose xmldiff
+      # TODO: replace with tox tests from #38
+      #- name: run nosetests
+      #  run: nosetests --with-coverage --cover-package=galaxyxml
+
+    - name: Run smoke test on example
+      run: |
+        python examples/example.py > tmp.xml
+        xmldiff tmp.xml examples/tool.xml
```
Comment (bernt-matthias) [inline .github/workflows/pr.yml]: While testing the other PR I had the impression that xmldiff just prints the diff but does not fail on a difference. So maybe 

```suggestion
        if xmldiff tmp.xml examples/tool.xml | grep '^'; then
            exit 1
        fi
```

(untested)

Comment (hexylena) [reply .github/workflows/pr.yml]: Fixed with `--check` flag, how convenient

**Comment by hexylena** [comment]
Merged despite failing, will fix that separately.

**Pull request merged**